File: tools/tests/fixtures/gen_el3/fire.go
package x
func StoreMsg(raw []byte, store KVStore) error {
    var msg Msg
    if err := proto.Unmarshal(raw, &msg); err != nil {
        return err
    }
    key := sha256.Sum256(raw)
    store.Set(key[:], raw)
    return nil
}
func Dedup(bz []byte, seen map[string]bool) bool {
    var m Msg
    proto.Unmarshal(bz, &m)
    if seen[string(bz)] {
        return true
    }
    seen[string(bz)] = true
    return false
}

File: tools/tests/fixtures/gen_el3/fire.rs
fn store_msg(raw: &[u8], store: &mut Store) -> Result<()> {
    let msg = Msg::try_from_slice(raw)?;
    let key = sha256(raw);
    store.set(key, raw);
    Ok(())
}
fn dedup(raw: &[u8], seen: &mut HashSet<Vec<u8>>) -> bool {
    let m = Msg::try_from_slice(raw).unwrap();
    if seen.contains(raw) { return true; }
    seen.insert(raw.to_vec());
    false
}
